core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncResponseTransformer.java

```diff
@@ -121,21 +121,26 @@ public void onNext(ByteBuffer byteBuffer) {
             if (byteBuffer == null) {
                 throw new NullPointerException("Element must not be null");
             }
-            
+
+            if (byteBuffer.remaining() == 0) {
```

Comment:
Is this just an optimization? The docs say the write finishes immediately if the buffer is empty

Reply:
I can update it, then. Missed that part of the docs.

Reply:
Fixed.